tracee-rules/signatures/rego/regosig/traceerego.go

```diff
@@ -183,24 +190,52 @@ func (sig *RegoSignature) OnEvent(e types.Event) error {
 	}
 
 	if len(results) > 0 && len(results[0].Expressions) > 0 && results[0].Expressions[0].Value != nil {
-		switch v := results[0].Expressions[0].Value.(type) {
-		case bool:
-			if v {
-				sig.cb(types.Finding{
-					Data:        nil,
-					Context:     ee,
-					SigMetadata: sig.metadata,
-				})
-			}
+
+		switch results[0].Expressions[0].Value.(type) {
 		case map[string]interface{}:
+			values, ok := results[0].Expressions[0].Value.(map[string]interface{})
+			if ok && len(values) <= 0 {
+				return nil
+			}
+		case bool:
+			sig.dispatch(results[0].Expressions[0].Value, ee)
+		}
+
+		// For AIO: result set can be an empty set of length=1, so we need to check value
+		values, ok := results[0].Expressions[0].Value.(map[string]interface{})
+		if ok && len(values) <= 0 {
+			return nil
+		}
+
+		sig.dispatch(values, ee)
+
+	}
+	return nil
+}
+
+func (sig *RegoSignature) dispatch(val interface{}, ee tracee.Event) {
```

Comment:
The tracee_match_all rule returns a map from signature id to values. Where do we have logic to iterate through that map and create 0 to many findings?
Do we have tests for that in aio_test.go?